pyproject.toml

```diff
@@ -9,25 +9,29 @@ license = "Apache-2.0"
 "napalm" = "nornir_napalm.plugins.connections:Napalm"
 
 [tool.poetry.dependencies]
-python = ">=3.7,<4.0"
-napalm = "^4"
+python = ">=3.8,<4.0"
+napalm = "^5"
 nornir = { version = "~3", allow-prereleases = true }
 
 [tool.poetry.dev-dependencies]
-black = "^22.6.0"
-mypy = "1.0.0"
+black = "^24"
```

Comment:
I pretty strongly think we should pin black, mypy, and pytest to exact versions as I don't want GH-actions to break just because of a `poetry update`.

All three of these tools are pretty annoying when you have to interrupt what you are doing to fix accidental breakage they caused (and these are all just `dev` dependencies i.e. doesn't affect general users and their installations).

Reply:
I totally understand. I pinned black, mypy, pytest, pytest-cov and nbval. At the moment, I am still enabling bugfixes with the `^24.3.0`, or should I pin it hard to `24.3.0`?

Reply:
I updated the PR with the exact pinning for black and mypy